### DISCUSSION TITLE
Update master server list

### DISCUSSION THREAD

**Discussion by Flaise** [Thinkofname/steamworks-rs#15] [Q&A]
I'm trying to verify that I'm setting up a game server correctly by checking the Steam client's built in server browser. After running the test repeatedly and clicking the [Refresh All] button repeatedly, I'm still not seeing anything added to the server browser.

Here's the code I'm running:

    #[test]
    fn steam_testing() {
        println!("  a");
        let (server, single) = Server::init(
            [0, 0, 0, 0].into(),
            23333, 23334, 23335,
            ServerMode::Authentication, "1.0.0"
        ).unwrap();
        println!("  b");

        server.set_product("Spacewar");
        server.set_game_description("basic server test");
        server.set_dedicated_server(true);
        server.log_on_anonymous();

        println!("  c {:?}", server.steam_id());

        let _cb = server.register_callback(|v: AuthSessionTicketResponse| {
            println!("Got response: {:?}", v.result)
        });
        let _cb = server.register_callback(|v: ValidateAuthTicketResponse| println!("{:?}", v));

        println!("  d");

        for _ in 0 .. 2000 {
            single.run_callbacks();
            ::std::thread::sleep(::std::time::Duration::from_millis(50));
        }

        println!("END");
    }

And here's the output:

    running 1 test
      a
    dlopen failed trying to load:
    steamclient.dylib
    with error:
    dlopen(steamclient.dylib, 2): image not found
    [S_API] SteamAPI_Init(): Loaded '/Users/flaise/Library/Application Support/Steam/Steam.AppBundle/Steam/Contents/MacOS/steamclient.dylib' OK.  (First tried local 'steamclient.dylib')
    CAppInfoCacheReadFromDiskThread took 4 milliseconds to initialize
    CApplicationManagerPopulateThread took 0 milliseconds to initialize (will have waited on CAppInfoCacheReadFromDiskThread)
    Setting breakpad minidump AppID = 480
      b
      c SteamId(90071992547409920)
      d
    test tests::steam_testing ... test tests::steam_testing has been running for over 60 seconds
    END
    test tests::steam_testing ... ok

I've tried changing the authentication mode, server version, product name, and commenting out the line that enables dedicated server mode. In all of these cases it continues to run without error but it still doesn't put anything in the server browser.

The only think I could think of was that [heartbeats](https://partner.steamgames.com/doc/api/ISteamGameServer#EnableHeartbeats) were disabled by default so I added the following to my local copy of the repository:

```
    // steamworks-sys/src/lib.rs
    pub fn SteamAPI_ISteamGameServer_EnableHeartbeats(instance: *mut ISteamGameServer, active: bool);
```
```
    // src/server.rs
    pub fn enable_heartbeats(&self, active: bool) {
        unsafe {
            sys::SteamAPI_ISteamGameServer_EnableHeartbeats(self.server, active);
        }
    }
```
```
    // unit test
    server.enable_heartbeats(true);
```
    
... And still nothing.

Do you have any idea of what I might be doing wrong?

**Comment by Thinkofname** [comment]
I haven't used the server browser feature yet. I've only used the server stuff for direct connections. 
The only thing I can think of is the query port not being publicly accessible?
